### ISSUE TITLE
Migration failure with latest PostgreSQL release 11.4

### ISSUE THREAD

**Issue by tarzan** [elixir-ecto/ecto_sql#123]
### Environment

* Elixir version (elixir -v): 1.8.1
* Database and version (PostgreSQL 9.4, MongoDB 3.2, etc.): PostgreSQL 11.4
* Ecto version (mix deps): 3.1.5
* Database adapter and version (mix deps): pg
* Operating system: MacOS Mojave 

### Current behavior
Considering this table
```elixir
defmodule MyApp.Repo.CreatePosts do
  use Ecto.Migration

  def change do
    create table(:likes) do
      add :user_id, references(:users, on_delete: :nothing)
      add :post_id, references(:posts, on_delete: :nothing)

      timestamps()
    end

    create index(:likes, [:user_id])
    create index(:likes, [:post_id])

    create unique_index(:likes, [:user_id, :post_id], name: :unique_user_post_index)
  end
end
```
and a follow-up migration that just aims to change the `on_delete` behaviour:
```elixir
defmoule MyApp.Repo.ChangeOnDeleteOfUsersAndPostsOnLikes do
   use Ecto.Migration

   def change do
     alter table(:likes) do
        modify :user_id, references(:users, on_delete: :delete_all), from: references(:users, on_delete: :nothing)
        modify :post_id, references(:posts, on_delete: :delete_all), from: references(:posts, on_delete: :nothing)
     end
   end
end
```
Now fails in PostgreSQL since 11.4, but worked fine in previous versions. The error message given is:
```
* (Postgrex.Error) ERROR 42P07 (duplicate_table) relation "unique_user_post_index" already exists
```

### Expected behavior
Not sure if this isn't *just* a bug introduced with this release of PostgreSQL or if this is a bug on ecto_sql. What I did find out however is that the following SQL is being generated:
```sql
ALTER TABLE "likes" 
DROP CONSTRAINT "likes_user_id_fkey", 
ALTER COLUMN "user_id" TYPE bigint, 
ADD CONSTRAINT "likes_user_id_fkey" FOREIGN KEY ("user_id") REFERENCES "users"("id") ON DELETE CASCADE, 
DROP CONSTRAINT "likes_post_id_fkey", 
ALTER COLUMN "post_id" TYPE bigint, 
ADD CONSTRAINT "likes_post_id_fkey" FOREIGN KEY ("post_id") REFERENCES "posts"("id") ON DELETE CASCADE
```
Which fails when run on the `psql` console as well, but the following SQL executes and has the desired result:
```sql
ALTER TABLE "likes" 
DROP CONSTRAINT "likes_user_id_fkey", 
ADD CONSTRAINT "likes_user_id_fkey" FOREIGN KEY ("user_id") REFERENCES "users"("id") ON DELETE CASCADE, 
DROP CONSTRAINT "likes_post_id_fkey", 
ADD CONSTRAINT "likes_post_id_fkey" FOREIGN KEY ("post_id") REFERENCES "posts"("id") ON DELETE CASCADE
```

So, the question is: Should a migration that changes/resets the foreign key reference on a column also declare the column type that doesn't change? I understand that changing the reference to another table might result in a change of type for the column and that it's kinda hard to figure out if this does beforehand. However, the required information is all there in the `from:` option....

**Comment by josevalim** [comment]
Thank you for the detailed wrap-up. Could you please report this bug in PostgreSQL, if they haven't reported yet? I would like to hear from them if it is not a regression on their side before we take any action.

**Comment by tarzan** [comment]
> Could you please report this bug in PostgreSQL

As far as I understand I have to report it through a mailinglist to which I need to be subscribed first, right? Are you or any of the other maintainers subscribed to this list?

**Comment by josevalim** [comment]
I am not sure about their process and I am not subscribed to their mailing list, sorry.

**Comment by tarzan** [comment]
For reference, I reported this on the psql bug mailing list, and this was the response:

> Yeah, this is an embarrassing bug in the fix for bug #15835. 
> A fix is already committed, 
> https://git.postgresql.org/gitweb/?p=postgresql.git&a=commitdiff&h=afaf48afb 
> and it will be in the next minor releases. 
> 
> regards, tom lane

**Comment by josevalim** [comment]
Great so I will close it on our side as it isn't an Ecto bug. Thanks for following up!
